### PR TITLE
feat: [CDS-85446]:tas-bg-ff-validation

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
1. Added new validation for tas bg step behind feature flag

relevant changes (pr): https://github.com/harness/harness-core-ui/pull/19128

Test cases:
1.validated from local env. ff true/false

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by litvinovstas** [comment]
trigger messagecheck

**Comment by litvinovstas** [comment]
trigger messagecheck

File: src/modules/75-cd/components/PipelineSteps/TasBGAppSetup/TasBGAppSetup.tsx [hunk 1]
```diff
@@ -125,13 +130,10 @@ export class TasBGAppSetupStep extends PipelineStep<TasBGAppSetupData> {
...
   }
 
-  validateInputSet({
-    data,
-    template,
-    getString,
-    viewType
-  }: ValidateInputSetProps<TasBGAppSetupData>): FormikErrors<TasBGAppSetupData> {
+  validateInputSet(props: ValidateInputSetProps<TasBGAppSetupData>): FormikErrors<TasBGAppSetupData> {
+    const { data, template, getString, viewType } = props
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/TasBGAppSetup/TasBGAppSetup.tsx:134]: It should come as new argument here. Editing template object would not be the right way to do it.

**Comment by litvinovstas** [comment]
trigger messagecheck

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineSteps/TasBGAppSetup/TasBGAppSetup.tsx (hunk 1 above)
Comment (litvinovstas) [reply src/modules/75-cd/components/PipelineSteps/TasBGAppSetup/TasBGAppSetup.tsx]: improved

File: src/modules/70-pipeline/components/AbstractSteps/Step.tsx [hunk 2]
```diff
@@ -31,12 +32,16 @@ export interface InputSetData<T> {
...
+export type ValidateInputSetFF = {
+  [key in FeatureFlag]?: boolean
+}
 export interface ValidateInputSetProps<T> {
   data: T
   template?: T
   getString?: UseStringsReturn['getString']
   viewType: StepViewType
   allValues?: T
+  validateFeatureFlags?: ValidateInputSetFF
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/AbstractSteps/Step.tsx:44]: nit: `featureFlags` or `featureFlagValues` is fine/better.

File: src/modules/70-pipeline/components/AbstractSteps/Step.tsx [hunk 3]
```diff
@@ -31,12 +32,16 @@ export interface InputSetData<T> {
   readonly?: boolean
 }
 
+export type ValidateInputSetFF = {
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/AbstractSteps/Step.tsx:35]: We can use `FeatureFlagMap` from `src/framework/AppStore/AppStoreContext.tsx` for type, no need to create new type.

**Pull request merged**